python/ray/data/_internal/planner/exchange/fast_repartition_task_scheduler.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+from ray.data._internal.execution.interfaces import RefBundle
+from ray.data._internal.planner.exchange.interfaces import ExchangeTaskScheduler
+from ray.data._internal.progress_bar import ProgressBar
+from ray.data._internal.remote_fn import cached_remote_fn
+from ray.data._internal.split import _split_at_indices
+from ray.data._internal.stats import StatsDict
+from ray.data.block import Block, BlockMetadata
+from ray.types import ObjectRef
+
+
+class FastRepartitionTaskScheduler(ExchangeTaskScheduler):
+    """
+    The fast (non-shuffle) repartition scheduler.
+
+    First, we calculate global splits needed to produce `output_num_blocks` blocks.
+    After the split blocks are generated accordingly, reduce tasks are scheduled
+    to combine split blocks together.
+    """
+
+    def execute(
+        self,
+        refs: List[RefBundle],
+        output_num_blocks: int,
+        map_ray_remote_args: Optional[Dict[str, Any]] = None,
+        reduce_ray_remote_args: Optional[Dict[str, Any]] = None,
+    ) -> Tuple[List[RefBundle], StatsDict]:
+        input_num_rows = 0
```

Comment:
the corresponding implementation is: https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/fast_repartition.py#L14